daemons/fenced/fenced_commands.c

```diff
@@ -1099,14 +1105,18 @@ dynamic_list_search_cb(GPid pid, int rc, const char *output, gpointer user_data)
 
     /* If we successfully got the targets earlier, don't disable. */
     if (rc != 0 && !dev->targets) {
-        crm_notice("Disabling port list queries for %s: %s "
+        if (g_hash_table_lookup(dev->params, PCMK_STONITH_HOST_LIST) == NULL &&
+            g_hash_table_lookup(dev->params, PCMK_STONITH_HOST_MAP) == NULL) {
+            /*
+                If the old code is correct, it will be changed to "status" in the case of "dynamic-list" 
+                and if neither pcmk_host_list nor pcmk_host_map is specified.
+            */
```

Comment:
I think I understand this code now. If host list or host map is *not* null, then the only way we could get to this function is if the user explicitly set the check type to list (since it would default to static-list in that situation). Therefore we can skip the fallback in that case, since the user explicitly requested list.

I think we can more directly express that with

    if (g_hash_table_lookup(dev->params, PCMK_STONITH_HOST_CHECK) == NULL) {

In other words: If we're here because the user didn't specify a check type, and we defaulted to list, but list didn't work, then fall back to status. That makes sense to me.

Reply:
Hi Ken,

I think your idea is a more concise expression.

What about the next fix?
I also changed the comment.

```
    if (rc != 0 && !dev->targets) {
        if (g_hash_table_lookup(dev->params, PCMK_STONITH_HOST_CHECK) == NULL {
            /*
                If the operation fails if the user does not explicitly specify "dynamic-list", it will fall back to "status".
            */
            crm_notice("Disabling port list queries for %s (%d): %s", dev->id, rc, output);
            /* Fall back to status */
            g_hash_table_replace(dev->params, strdup(STONITH_ATTR_HOSTCHECK), strdup("status"));
        }
(snip)
````



Many thanks,
Hideo Yamauchi.

Reply:
Hi Hideo, that works for me.

Reply:
Hi Ken,

Okay!

thanks!